sound/soc/soc-pcm.c

```diff
@@ -1428,7 +1433,8 @@ static bool dpcm_end_walk_at_be(struct snd_soc_dapm_widget *widget,
 
 	if (dir == SND_SOC_DAPM_DIR_OUT) {
 		for_each_card_rtds(card, rtd) {
-			if (!rtd->dai_link->no_pcm)
+			if (!rtd->dai_link->no_pcm ||
+			    !strcmp(widget->name, SOC_VIRT_DAI_PLAYBACK))
```

Comment:
Maybe better to use a bool flag here like no_pcm ? This should be a seperate patch too btw.

Reply:
@lgirdwood I'll have to work more on making a meaningful split of the patches, but so far it seems to me, that there will anyway be a relatively big core ALSA / SOF VirtIO patch still. It will contain both the VirtIO (guest) driver itself and supporting code like the above, because the supporting code makes no sense without the driver files, and driver files are non-functional or potentially even crashing without supporting code. Maybe this will become a bool flag in the end, yes.

Reply:
This is another of those string matching locations.